src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -210,10 +210,17 @@ CORINFO_CLASS_HANDLE Compiler::gtGetStructHandleForHWSIMD(var_types simdType, Co
 //
 /* static */ bool Compiler::vnEncodesResultTypeForHWIntrinsic(NamedIntrinsic hwIntrinsicID)
 {
+    // No extra type information is needed for scalar HW Intrinsic.
+    //
+    if (HWIntrinsicInfo::lookupCategory(hwIntrinsicID) == HW_Category_Scalar)
+    {
+        return false;
+    }
```

Comment:
We may also want a path for `HW_Category_Special` or maybe better check for:
```suggestion
    unsigned simdSize = 0;
    if (HWIntrinsicInfo::tryLookupSimdSize(hwIntrinsicID, &simdSize) && (simdSize == 0))
    {
        return false;
    }
```

All scalar intrinsics, regardless of category, should have a `simdSize == 0`. Likewise, all SIMD intrinsics should have a size of `8`, `16`, `32`, or `-1` (where `-1` implies the intrinsic supports multiple SIMD sizes and is not a scalar).

Reply:
I think the current check is "accurate enough" today since the few "special" scalars tend to have all `INS_invalid` entries and are specially handled in codegen instead, but its probably better to be more explicit here.

There's also a consideration for the type encoding that certain special or helper intrinsics might specify `INS_invalid` and get specialized handling in lowering/codegen and so it might be important to have the type anyways. I'm unsure if there is any actual issue here or not, but its probably worth auditing at some point in the future.

Reply:
I agree, except that it doesn't set the `simdSize` if it was `-1`. So probably I should initialize `simdSize` to something like `100` or something and then see if became `0`. Wil fix it.

Reply:
That's what the `tryLookupSimdSize` check was for. It should return `false` if `simdSize == -1` and so the second check (`simdSize == 0`) won't get hit.

Reply:
ah. was an oversight on my part.